### DISCUSSION TITLE
Arbitrary JSON result?

### DISCUSSION THREAD

**Discussion by cprieto** [poem-web/poem#599] [Q&A]
I have an endpoint that can return a JSON array of objects, but the schema of the object can vary. I have no idea how to represent this in Poem OpenAPI, any suggestion will be welcomed!

**Comment by sunli829** [comment]
define a union object

https://docs.rs/poem-openapi/2.0.27/poem_openapi/derive.Union.html

**Reply by cprieto** [reply]
Nah, in a union object you know the types or shape of the result, arbitrary means it could have any shape. My impression is that I will have to define a `Type` for this, I will try and report back.

**Reply by sunli829** [reply]
Well, maybe what you want is actually this.

https://docs.rs/poem-openapi/2.0.27/poem_openapi/types/struct.Any.html
